tests/core/providers/test_http_request_retry.py

```diff
@@ -0,0 +1,183 @@
+import pytest
+from unittest.mock import (
+    patch,
+)
+
+import aiohttp
+from requests.exceptions import (
+    ConnectionError,
+    HTTPError,
+    Timeout,
+    TooManyRedirects,
+)
+
+from web3 import (
+    AsyncHTTPProvider,
+    AsyncWeb3,
+    Web3,
+)
+from web3.providers import (
+    HTTPProvider,
+    IPCProvider,
+)
+from web3.providers.rpc.utils import (
+    ExceptionRetryConfiguration,
+    check_if_retry_on_failure,
+)
+from web3.types import (
+    RPCEndpoint,
+)
+
+TEST_RETRY_COUNT = 3
+
+
+@pytest.fixture
+def w3():
+    errors = (ConnectionError, HTTPError, Timeout, TooManyRedirects)
+    config = ExceptionRetryConfiguration(errors=errors, retries=TEST_RETRY_COUNT)
+    return Web3(HTTPProvider(exception_retry_configuration=config))
+
+
+def test_default_request_retry_configuration_for_http_provider():
+    w3 = Web3(HTTPProvider())
+    assert w3.provider.exception_retry_configuration == ExceptionRetryConfiguration()
+
+
+def test_check_if_retry_on_failure_false():
+    methods = [
+        "eth_sendTransaction",
+        "personal_signAndSendTransaction",
+        "personal_sendTransaction",
+    ]
+
+    for method in methods:
+        assert not check_if_retry_on_failure(method)
+
+
+def test_check_if_retry_on_failure_true():
+    method = "eth_getBalance"
+    assert check_if_retry_on_failure(method)
+
+
+@patch("web3.providers.rpc.rpc.make_post_request", side_effect=ConnectionError)
+def test_check_send_transaction_called_once(make_post_request_mock, w3):
+    with pytest.raises(ConnectionError):
+        w3.provider.make_request(
+            RPCEndpoint("eth_sendTransaction"), [{"to": f"0x{'00' * 20}", "value": 1}]
+        )
+    assert make_post_request_mock.call_count == 1
+
+
+@patch("web3.providers.rpc.rpc.make_post_request", side_effect=ConnectionError)
+def test_valid_method_retried(make_post_request_mock, w3):
+    with pytest.raises(ConnectionError):
+        w3.provider.make_request(RPCEndpoint("eth_getBalance"), [f"0x{'00' * 20}"])
+    assert make_post_request_mock.call_count == TEST_RETRY_COUNT
+
+
+def test_exception_retry_config_is_strictly_on_http_provider():
```

Comment:
Could do WS, and AsyncWeb3 versions too, for completeness. But a take/leave as you like.

Reply:
Yeah I just ported the test as it was over here. Refraining from adding too many additional things at the moment but this seems simple enough. I'll put it in 👍🏼